csharp/src/Drivers/Apache/Spark/SparkDatabricksReader.cs

```diff
@@ -32,51 +35,64 @@ internal sealed class SparkDatabricksReader : IArrowArrayStream
         {
             this.statement = statement;
             this.schema = schema;
+            this.Trace = trace;
         }
 
+        private ActivityTrace Trace { get; }
+
         public Schema Schema { get { return schema; } }
 
         public async ValueTask<RecordBatch?> ReadNextRecordBatchAsync(CancellationToken cancellationToken = default)
         {
-            while (true)
+            return await Trace.TraceActivityAsync(async (activity) =>
             {
-                if (this.reader != null)
+                while (true)
                 {
-                    RecordBatch? next = await this.reader.ReadNextRecordBatchAsync(cancellationToken);
-                    if (next != null)
+                    if (this.reader != null)
                     {
-                        return next;
+                        RecordBatch? next = await this.reader.ReadNextRecordBatchAsync(cancellationToken);
+                        if (next != null)
+                        {
+                            return next;
+                        }
+                        this.reader = null;
                     }
-                    this.reader = null;
-                }
 
-                if (this.batches != null && this.index < this.batches.Count)
-                {
-                    this.reader = new ArrowStreamReader(new ChunkStream(this.schema, this.batches[this.index++].Batch));
-                    continue;
-                }
+                    if (this.batches != null && this.index < this.batches.Count)
+                    {
+                        this.reader = new ArrowStreamReader(new ChunkStream(this.schema, this.batches[this.index++].Batch));
+                        continue;
+                    }
 
-                this.batches = null;
-                this.index = 0;
+                    this.batches = null;
+                    this.index = 0;
 
-                if (this.statement == null)
-                {
-                    return null;
-                }
+                    if (this.statement == null)
+                    {
+                        return null;
+                    }
 
-                TFetchResultsReq request = new TFetchResultsReq(this.statement.OperationHandle, TFetchOrientation.FETCH_NEXT, this.statement.BatchSize);
-                TFetchResultsResp response = await this.statement.Connection.Client!.FetchResults(request, cancellationToken);
-                this.batches = response.Results.ArrowBatches;
+                    activity?.AddEvent("db.operation.name.start", [new("db.operation.name", nameof(Client.FetchResults))]);
+                    TFetchResultsReq request = new TFetchResultsReq(this.statement.OperationHandle, TFetchOrientation.FETCH_NEXT, this.statement.BatchSize);
+                    TFetchResultsResp response = await this.statement.Connection.Client!.FetchResults(request, cancellationToken);
+                    activity?.AddEvent("db.operation.name.end",
```

Comment:
Helps to trace the begin/end of fetching results from the server so the duration can be calculated from the difference.